src/screen.ml

```diff
@@ -14,6 +14,9 @@ let create (width : int) (height : int) (scale : int) (palette : Palette.t) : t
   if height <= 0 then raise (Invalid_argument "Invalid height");
   { width ; height ; scale ; palette ; font = None}
 
+let update (screen : t) (new_palette : Palette.t) : t =
```

Comment:
Screen has things other than the palette on it, so we should either use a named parameter here to make it clear you're updating the palette, or name the function something that indicates it is the palette updating. It might be one day, for example, we want to update the font or the screen resolution.